apps/ledger-live-mobile/src/screens/FirmwareUpdate/index.tsx

```diff
@@ -537,37 +553,46 @@ export const FirmwareUpdate = ({
       );
     }
 
-    const error =
-      updateActionState.error ??
-      restoreAppsState.error ??
-      installLanguageState.error ??
-      staxLoadImageState.error ??
-      staxFetchImageState.error;
-
-    // a TransportRaceCondition error is to be expected since we chain multiple
-    // device actions that use different transport acquisition paradigms
-    // the action should, however, retry to execute and resolve the error by itself
-    // no need to present the error to the user
-    if (
-      error &&
-      !["TransportRaceCondition", "LockedDeviceError", "UnresponsiveDeviceError"].includes(
-        error.name,
-      )
-    ) {
+    /**
+     * Finds the user-solvable error from the (current) fw update step, if there is one.
+     * If there is an error during the current fw update step but it is not user-solvable,
+     * then either: this current fw update step is skipped or this error is ignored
+     * (logic handled in `useUpdateFirmwareAndRestoreSettings`).
+     * And in both cases: nothing is displayed to the user (logs are saved from the hook).
+     *
+     * Especially: a `TransportRaceCondition` error is to be expected since we chain multiple
+     * device actions that use different transport acquisition paradigms the action should,
+     * however, retry to execute and resolve the error by itself.
+     * There is no need to present the error to the user.
+     */
+    const userSolvableError = hasUserSolvableErrors
```

Comment:
Shouldn't this be defined in the same place as `hasUserSolvableErrors` ? i.e inside the hook.
Find a potential `userSolvableError` by looping through the potential errors and for each check if it's an instance of one of the user solvable error classes.
That way you only only define each array once and you prevent forgetting checking one of the many types of errors possible.

So in the hook:
```ts
const userSolvableError = [
  connectManagerState.error,
  updateActionState.error,
  restoreAppsState.error,
  installLanguageState.error,
  staxLoadImageState.error,
  staxFetchImageState.error,
].find(
  err => userSolvableErrors.some(errorClass => err instanceof errorClass)
)

return {
  userSolvableError,
  hasUserSolvableError: Boolean(userSolvableError),
};
```

Reply:
Yes good idea ;)